Lib/asyncio/streams.py

```diff
@@ -35,6 +35,9 @@ def __init__(self, partial, expected):
         self.partial = partial
         self.expected = expected
 
+    def __reduce__(self):
+        return type(self), (self.partial, self.expected)
```

Comment:
There is other problem with this solution -- it loses all additional attributes.

Reply:
asyncio doesn't set any additional attributes, so I think it's OK

Reply:
What if somebody will create a subclass?

Adding `self.__dict__` as the third component will save all attributes. But not slots.

bpo-26579 could help to support general cases. But pickling exceptions is a headache in any case.

Reply:
> What if somebody will create a subclass?

Well, they then can handle pickling on their own.  I mean in asyncio we control both exceptions and don't subclass them.  If someone outside of asyncio decides to subclass or otherwise reuse these exceptions, it's entirely up to them.

I don't want to introduce any extra overhead here.

Reply:
Then everything is fine. Except misleading repr, but this is another story.